### PR TITLE
Document node's store structure

### LINKED ISSUE #1341: docs: Document the node's store structure clearly
Not sure if this is already done, but I looked through docs and was unable to find.

We should document the node's datastore and its contents.

### PR DESCRIPTION
**Pull request by Mackenzie-OO7**
## Overview

closes #1341 

## Summary by CodeRabbit

- **New Features**
  - Added "Datastore structure" link to the sidebar for improved navigation in the documentation.
  - Introduced a detailed overview of the Celestia node's datastore structure in the how-to guide, enhancing user understanding of data management.

### CODE REVIEW

**Comment by jcstein** [comment]
@coderabbitai review

File: nodes/celestia-node-store-structure.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  - `app.toml`: The configuration file to define application settings such as logging levels and performance tuning.
+
+  - `client.toml`: used to configure parameters for client interactions like API endpoints or timeouts.
+
+- `config.toml`: This is the node's primary configuration file.
+It defines the node's core settings such as the network parameters and the node's identity.
+
+- `inverted_index`: This file stores indexed data that can be used to make blockchain queries.
+
+- `/keys`: This directory stores the cryptographic key pair that is used to operate the node.
```
Comment (jcstein) [inline nodes/celestia-node-store-structure.md:31]:
```suggestion
- `/keys`: This directory stores the cryptographic key pairs that are used to operate the node.
```

File: nodes/celestia-node-store-structure.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+- `/config`: This directory stores the files used to set the node's operational parameters.
+It contains the following configuration files:
+
+  - `app.toml`: The configuration file to define application settings such as logging levels and performance tuning.
+
+  - `client.toml`: used to configure parameters for client interactions like API endpoints or timeouts.
+
+- `config.toml`: This is the node's primary configuration file.
+It defines the node's core settings such as the network parameters and the node's identity.
```
Comment (jcstein) [inline nodes/celestia-node-store-structure.md:27]: "and the node's identity" is this true @renaynay ?

**Review by distractedm1nd — CHANGES_REQUESTED** [review]
Hey, thanks for this PR, but there are some confusing errors...

There is no app.toml or client.toml, or /config. Was this PR AI generated?

**Comment by Mackenzie-OO7** [comment]
> Hey, thanks for this PR, but there are some confusing errors...
> 
> There is no app.toml or client.toml, or /config. Was this PR AI generated?

Hi, I followed the draft in the linked issue as a guide. Is there somewhere else I could reference?

**Comment by jcstein** [comment]
> draft in the linked issue as a guide
yeah, I remember writing this but don't know where the extra directories and files are from. Are there more with full and bridge nodes? @Mackenzie-OO7 you can figure out the node store structure by initializing each node type `celestia bridge init` etc. and looking at the node store created. here's celestia light for example:

```bash
$ tree .celestia-light
.celestia-light
├── config.toml
├── data
│   ├── 000002.vlog
│   ├── 00022.mem
│   ├── 001174.sst
│   ├── DISCARD
│   ├── KEYREGISTRY
│   ├── LOCK
│   └── MANIFEST
└── keys
    ├── NJ3XILLTMVRXEZLUFZVHO5A
    ├── OAZHALLLMV4Q
    └── keyring-test
        ├── 968218...32d7c.address
        └── my_celes_key.info

4 directories, 129 files
```

**Comment by Mackenzie-OO7** [comment]
> 
> yeah, I remember writing this but don't know where the extra directories and files are from. Are there more with full and bridge nodes? @Mackenzie-OO7 you can figure out the node store structure by initializing each node type `celestia bridge init` etc. and looking at the node store created. here's celestia light for example:
> 

Ohh. Thanks!😅

**Comment by jcstein** [comment]
hey, @distractedm1nd could you please help summarize what these are? thanks!

**Comment by distractedm1nd** [comment]
Belongs to badger, I have a general idea of what LOCK does but not the others - but i don't recommend describing these any further than "Various files belonging to Node LSM storage system"

**Comment by jcstein** [comment]
Awesome, thanks @distractedm1nd. let's use "Various files belonging to Node LSM storage system" for all of those @Mackenzie-OO7

**Comment by jcstein** [comment]
Kind bump here @Mackenzie-OO7 !

> Awesome, thanks @distractedm1nd. let's use "Various files belonging to Node LSM storage system" for all of those @Mackenzie-OO7

File: nodes/celestia-node-store-structure.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+The node's datastore refers to the storage structure
+used to manage the data that supports the node's operation.
+It consists of directories and files that contain the node's state,
+configuration, and other information relevant to the node.
+
+The following are the directories and files found in the datastore:
+
+- `/blocks`: This directory stores blocks. Each file contained in this directory
+represents a block on Celestia and contains its associated data. However, the light node's datastore does not contain this directory, as light nodes do not store blocks.
```
Comment (jcstein) [inline nodes/celestia-node-store-structure.md:15]: - [ ] Add clarification that it is present on bridge and full nodes, but not light nodes

File: nodes/celestia-node-store-structure.md [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+- `/data`: This directory contains block headers and various files belonging to node LSM storage system such as `DISCARD`, `KEYREGISTRY`, and `MANIFEST`.
+
+- `/index`: This directory stores the index files that handle mapping specific keys such as block heights, to the corresponding data. Similar to `/blocks`, the light node's datastore does not include this directory, as they do not perform indexing.
+
+- `/inverted_index`: This directory stores the inverted index files used for mapping queries to the corresponding data location, and various files belonging to node LSM storage system, such as `DISCARD`, `KEYREGISTRY`, `LOCK`, and `MANIFEST`. The light node's datastore does not contain this directory.
```
Comment (jcstein) [inline nodes/celestia-node-store-structure.md:21]: - [ ] define "LSM" before using it as acronym

**Review by jcstein — CHANGES_REQUESTED** [review]

**Comment by jcstein** [comment]
let's close this and rework it into the new structure. wdyt? also after reviewing, I think a lot of what is maybe AI generated here needs to go. many of these aren't in my node store...

**Comment by jcstein** [comment]
makes sense! thank you for clarifying

**Comment by jcstein** [comment]
didn't fully resolve in that PR! :sweat_smile:

File: how-to-guides/celestia-node-store-structure.md [hunk 5]
```diff
@@ -0,0 +1,28 @@
+---
+description: This section contains information on the celestia-node datastore and its contents.
```
Comment (jcstein) [inline how-to-guides/celestia-node-store-structure.md:2]: does the navigation flow from previous actual page (not external link) and to the next page work as expected when you run this locally?

**Review by jcstein — APPROVED** [review]

**Pull request merged**